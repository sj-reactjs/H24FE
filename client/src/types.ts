export type Category = {
    name: string
    categoryArticles: CategoryArticle
    articleCount: number
    childrenCategories: ChildCategory[]
}

export type Article = {
    name: string
    variantName: string
    prices: Prices
    images: Image[]
}

export type ChildCategory = {
    name: string
    urlPath: string
}

export type Prices = {
    currency: string
    regular: {
        value: number
    }
}

export type Image = {
    path: string
}

export type CategoryArticle = {
    articles: Article[]
}

export type PageProps = {
    categoryArticles: CategoryArticle
    articleCount: number
    childrenCategories: ChildCategory[]
    locale: string;
    name: string;
}

export type SidebarProps = {
    categories: ChildCategory[];
};