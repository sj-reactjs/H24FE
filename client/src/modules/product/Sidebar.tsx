import React from 'react'
import Box from '../../components/Box';
import List, { ListItem } from '../../components/List';
import { ExternalLink } from '../../components/ExternalLink';
import { ChildCategory, SidebarProps } from '../../types'

const Sidebar: React.FC<SidebarProps> = ({ categories }) => {
    return (
        <Box
            borderRadius="md"
            borderColor={'primary_600'}
            p={'1rem'}
            marginBottom={`1em`}
            boxShadow="lg"
            className={'sidebar'}>
            <h3>Kategorien</h3>
            <List>
                {categories.map(({ name, urlPath }: ChildCategory, index: number) => <CategoryItem key={index} name={name} urlPath={urlPath} />)}
            </List>

        </Box>
    )
}

const CategoryItem: React.FC<ChildCategory> = ({ name, urlPath }) => {
    return (
        <ListItem aria-label={name}>
            <ExternalLink href={`/${urlPath}`}>{name}</ExternalLink>
        </ListItem>
    );
}

export default Sidebar;
