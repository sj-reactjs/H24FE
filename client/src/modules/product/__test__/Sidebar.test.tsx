import * as React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import Sidebar from '../Sidebar'
import { SidebarProps } from '../../../types'


describe('Sidebar Component', () => {
  let props: SidebarProps;
  beforeEach(() => {
    props = {
      categories: [
        { "name": "Test1", "urlPath": "kategorie/t1/" },
        { "name": "Test2", "urlPath": "kategorie/t2/" },
        { "name": "Test3", "urlPath": "kategorie/t3/" }
      ],
    }
  })

  test('render', () => {
    const { queryByText } = render(<Sidebar categories={props.categories} />);
    const linkElement = queryByText(/Test1/i);
    expect(linkElement).toBeInTheDocument();
    const anotherLinkElement = queryByText(/Test10/i);
    expect(anotherLinkElement).not.toBeInTheDocument();
  })

  test('Component snapshot', () => {
    const { asFragment } = render(<Sidebar categories={props.categories} />);
    expect(asFragment()).toMatchSnapshot();
  })
})
