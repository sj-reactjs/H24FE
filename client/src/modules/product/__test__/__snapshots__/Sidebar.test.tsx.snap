// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component Component snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ePzfKO sidebar"
  >
    <h3>
      Kategorien
    </h3>
    <ul
      class="sc-dkPtRN lkGMbd"
      width="100%"
    >
      <li
        aria-label="Test1"
        class="sc-gsDKAQ dnEoat"
      >
        <a
          class="sc-hKwDye bLYwem"
          href="/kategorie/t1/"
        >
          Test1
        </a>
      </li>
      <li
        aria-label="Test2"
        class="sc-gsDKAQ dnEoat"
      >
        <a
          class="sc-hKwDye bLYwem"
          href="/kategorie/t2/"
        >
          Test2
        </a>
      </li>
      <li
        aria-label="Test3"
        class="sc-gsDKAQ dnEoat"
      >
        <a
          class="sc-hKwDye bLYwem"
          href="/kategorie/t3/"
        >
          Test3
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
